src/nsidc/metgen/netcdf_reader.py

```diff
@@ -0,0 +1,97 @@
+import json
+import os.path
+import xarray as xr
+import re
+from datetime import datetime, timezone
+
+from pyproj import CRS
+from pyproj import Transformer
+
+
+def extract_metadata(netcdf_path):
+    """
+    Read the content at netcdf_path and return a structure with temporal coverage
+    information, spatial coverage information, file size, and production datetime.
+    """
+
+    # TODO: handle errors if any needed attributes don't exist.
+    netcdf = xr.open_dataset(netcdf_path, decode_coords="all")
+
+    return { 
+        'size_in_bytes': os.path.getsize(netcdf_path),
+        'production_date_time': ensure_iso(netcdf.attrs['date_modified']),
+        'temporal': time_range(netcdf),
+        'geometry': {'points': json.dumps(spatial_values(netcdf))}
+    }
+
+def time_range(netcdf):
+    """Return an array of datetime strings"""
+    datetimes = []
+    datetimes.append(ensure_iso(netcdf.attrs['time_coverage_start']))
+    datetimes.append(ensure_iso(netcdf.attrs['time_coverage_end']))
+
+    return datetimes
+
+def spatial_values(netcdf):
+    """
+    Return an array of dicts, each dict representing one lat/lon pair like so:
+
+        {
+            "Longitude: float,
+            "Latitude: float
+        }
+    Eventually this can/should be pulled out of the netCDF-specific code into a
+    general-use module.
+    """
+
+    data_crs = CRS.from_wkt(netcdf.crs.crs_wkt)
+    crs_4326 = CRS.from_epsg(4326)
+    xformer = Transformer.from_crs(data_crs, crs_4326, always_xy=True)
+
+    # Adding padding should give us values that match up to the netcdf.attrs.geospatial_bounds
+    pad = abs(float(netcdf.crs.GeoTransform.split()[1]))/2
+    xdata = list(map(lambda x: x - pad if x < 0 else x + pad, netcdf.x.data))
+    ydata = list(map(lambda y: y - pad if y < 0 else y + pad, netcdf.y.data))
```

Comment:
See my other comment about using a list-comprehension instead of `map`. I forgot to say there that there is some Python Lore that suggests avoiding map in favor of list-comprehensions, but I don't remember exactly *why*. One reason is that they can sometimes be more compact and/or readable. E.g.:
`[x - pad if x < 0 else x + pad for x in netcdf.x.data]`

Reply:
I am all for the list comprehension approach -- my brain isn't yet making the Python idiom leap in any sort of automated way!